domain/autocert/state/state.go

```diff
@@ -85,7 +86,8 @@ WHERE  name = $M.name`
 		return errors.Trace(tx.Query(ctx, s, sqlair.M{"name": name}).Get(&row))
 	}); err != nil {
 		if err.Error() == sql.ErrNoRows.Error() {
-			return nil, errors.Annotatef(errors.NotFound, "autocert %s", name)
+			return nil, autocert.ErrCacheMiss
```

Comment:
I think this is the wrong place to place this.

My reasoning is that it's not the state layer that is implementing the autocert cache interface. It is the service that is implementing this. I think returning NotFound here is the correct response.

Then in the service layer we can do a `if errors.Is(err, errors.NotFound) {`

Reply:
yeah, I had think about this and then went on for state. Moved it to service layer